gateway/src/http_entrypoints.rs

```diff
@@ -516,14 +571,66 @@ async fn sp_component_update(
     rqctx: Arc<RequestContext<Arc<ServerContext>>>,
     path: Path<PathSpComponent>,
     body: TypedBody<UpdateBody>,
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    let comms = Arc::clone(&rqctx.context().sp_comms);
+    let PathSpComponent { sp, component } = path.into_inner();
+    let component = component_from_str(&component)?;
+    let UpdateBody { id, image, slot } = body.into_inner();
+
+    comms
+        .start_update(sp.into(), component, id, slot, image)
+        .await
+        .map_err(http_err_from_comms_err)?;
+
+    Ok(HttpResponseUpdatedNoContent {})
+}
+
+/// Get the status of an update being applied to an SP component
+///
+/// Getting the status of an update to the SP itself is done via the component
+/// name `sp`.
+#[endpoint {
+    method = GET,
+    path = "/sp/{type}/{slot}/component/{component}/update-status",
+}]
+async fn sp_component_update_status(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path: Path<PathSpComponent>,
+) -> Result<HttpResponseOk<Option<SpUpdateStatus>>, HttpError> {
+    let comms = Arc::clone(&rqctx.context().sp_comms);
+    let PathSpComponent { sp, component } = path.into_inner();
+    let component = component_from_str(&component)?;
+
+    let status = comms
+        .update_status(sp.into(), component)
+        .await
+        .map_err(http_err_from_comms_err)?;
+
+    Ok(HttpResponseOk(status.map(Into::into)))
+}
+
+/// Abort any in-progress update an SP component
+///
+/// Aborting an update to the SP itself is done via the component name `sp`.
+///
+/// If this returns successfully, it does not mean an update was actually
+/// aborted; it only means no update is currently in progress.
```

Comment:
On a successful return either the update corresponding to the given UUID has been aborted or it has already been applied. It will not be in progress or in an otherwise transient state

.... not sure that's better ;-)

Reply:
Could we get away with just this?

> On a successful return, the update corresponding to the given UUID will no longer be in progress.

Reply:
Perhaps "... no longer be in progress (either aborted or applied)."?